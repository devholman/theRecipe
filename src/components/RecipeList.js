import React from 'react';
import propTypes from 'prop-types';
import Recipe from './RecipeListItem';

const RecipeList = ({ recipes, style, favorites, ...props }) => {
  const allRecipes = recipes.map(recipe => {
    return (
      <Recipe
        key={recipe.id}
        recipe={recipe}
        favorited={favorites.includes(recipe.id)}
        {...props}
      />
    );
  });
  return <ul className="list-reset">{allRecipes}</ul>;
};

RecipeList.propTypes = {
  recipes: propTypes.array,
  favorites: propTypes.array,
  style: propTypes.object
};

RecipeList.defaultProps = {
  recipes: [],
  favorites: []
};
export default RecipeList;
