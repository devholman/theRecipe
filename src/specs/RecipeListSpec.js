import React from 'react';
import renderer from 'react-test-renderer';
import RecipeList from '../components/RecipeList';
const testRecipes = [
  {
    id: 1,
    name: 'recipe 1',
    category: 'cat 1'
  },
  {
    id: 2,
    name: 'recipe 2',
    category: 'cat 2'
  },
  {
    id: 3,
    name: 'recipe 3',
    category: 'cat 3'
  }
];
describe('<RecipeList />', () => {
  test('should not break when no recipes are passed', () => {
    const component = renderer.create(<RecipeList />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should render correctly', () => {
    const component = renderer.create(<RecipeList recipes={testRecipes} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should render favorites', () => {
    const component = renderer.create(
      <RecipeList recipes={testRecipes} favorites={[1]} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
