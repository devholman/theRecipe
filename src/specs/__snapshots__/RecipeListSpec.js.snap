// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeList /> should not break when no recipes are passed 1`] = `
<ul
  className="list-reset"
/>
`;

exports[`<RecipeList /> should render correctly 1`] = `
<ul
  className="list-reset"
>
  <li
    className="border-bottom border-bottom-dashed py2 pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      🗃️
    </span>
    <span>
      recipe 1
    </span>
    <span>
      cat 1
    </span>
  </li>
  <li
    className="border-bottom border-bottom-dashed py2 pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      🗃️
    </span>
    <span>
      recipe 2
    </span>
    <span>
      cat 2
    </span>
  </li>
  <li
    className="border-bottom border-bottom-dashed py2 pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      🗃️
    </span>
    <span>
      recipe 3
    </span>
    <span>
      cat 3
    </span>
  </li>
</ul>
`;

exports[`<RecipeList /> should render favorites 1`] = `
<ul
  className="list-reset"
>
  <li
    className="border-bottom border-bottom-dashed py2 pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      😀
    </span>
    <span>
      recipe 1
    </span>
    <span>
      cat 1
    </span>
  </li>
  <li
    className="border-bottom border-bottom-dashed py2 pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      🗃️
    </span>
    <span>
      recipe 2
    </span>
    <span>
      cat 2
    </span>
  </li>
  <li
    className="border-bottom border-bottom-dashed py2 pointer"
    onClick={[Function]}
  >
    <span
      aria-label="favorite"
      className="mr1"
      onClick={[Function]}
      role="img"
    >
      🗃️
    </span>
    <span>
      recipe 3
    </span>
    <span>
      cat 3
    </span>
  </li>
</ul>
`;
